### ISSUE TITLE
`generator-star-spacing`: add support for distinguishing class methods and property function shorthand

### ISSUE THREAD

**Issue by A-kirami** [eslint-stylistic/eslint-stylistic#951]
### Clear and concise description of the problem

I want the `generator-star-spacing` rule to distinguish between class methods and property function shorthand so that I can configure different spacing preferences for these two distinct function types. Currently, both are controlled by the same `method` override, which limits flexibility in code style configuration.

### Suggested solution

In rule `generator-star-spacing`, we could provide an additional `shorthand` configuration option alongside the existing `named`, `anonymous`, and `method` overrides. This would allow separate control for property function shorthand while maintaining backward compatibility.

Example configuration:

```js
/* eslint @stylistic/generator-star-spacing: ["error", {
    "before": false,
    "after": true,
    "method": "after",
    "shorthand ": "neither"
}] */

class Class { static* method() {} }

var shorthand = { *generator() {} };
```

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue
